code/modules/power/supermatter/supermatter.dm

```diff
@@ -1,438 +1,876 @@
+	integrity = round(100 - integrity * 100, 0.01)
+	integrity = integrity < 0 ? 0 : integrity
+	return integrity
 
-	damage = max(0, damage + between(-DAMAGE_RATE_LIMIT, (removed.temperature - CRITICAL_TEMPERATURE) / 150, damage_inc_limit))
+/obj/machinery/power/supermatter_crystal/proc/countdown()
+	set waitfor = FALSE
 
-	//Maxes out at 100% oxygen pressure
-	oxygen = clamp((removed.oxygen - (removed.nitrogen * NITROGEN_RETARDATION_FACTOR)) / removed.total_moles(), 0, 1)
+	if(final_countdown) // We're already doing it go away
+		return
+	final_countdown = TRUE
+
+	var/image/causality_field = image(icon, null, "causality_field")
+	add_overlay(causality_field, TRUE)
+
+	var/speaking = "[emergency_alert] The supermatter has reached critical integrity failure. Emergency causality destabilization field has been activated."
+	radio.autosay(speaking, name, null, list(z))
+	for(var/i in SUPERMATTER_COUNTDOWN_TIME to 0 step -10)
+		if(damage < explosion_point) // Cutting it a bit close there engineers
+			radio.autosay("[safe_alert] Failsafe has been disengaged.", name, null, list(z))
+			cut_overlay(causality_field, TRUE)
+			final_countdown = FALSE
+			return
+		else if((i % 50) != 0 && i > 50) // A message once every 5 seconds until the final 5 seconds which count down individualy
+			sleep(10)
+			continue
+		else if(i > 50)
+			speaking = "[DisplayTimeText(i, TRUE)] remain before causality stabilization."
+		else
+			speaking = "[i*0.1]..."
+		radio.autosay(speaking, name, null, list(z))
+		sleep(10)
+
+	explode()
+
+/obj/machinery/power/supermatter_crystal/proc/explode()
+	for(var/mob in GLOB.alive_mob_list)
+		var/mob/living/L = mob
+		if(istype(L) && L.z == z)
+			if(ishuman(mob))
+				//Hilariously enough, running into a closet should make you get hit the hardest.
+				var/mob/living/carbon/human/H = mob
+				H.hallucination += max(50, min(300, DETONATION_HALLUCINATION * sqrt(1 / (get_dist(mob, src) + 1))))
+			var/rads = DETONATION_RADS * sqrt(1 / (get_dist(L, src) + 1))
+			L.apply_effect(rads, IRRADIATE)
 
-	var/temp_factor
-	var/equilibrium_power
-	if(oxygen > 0.8)
-		//If chain reacting at oxygen > 0.8, we want the power at 800 K to stabilize at a power level of 400
-		equilibrium_power = 400
-		icon_state = "[base_icon_state]_glow"
-	else
-		//Otherwise, we want the power at 800 K to stabilize at a power level of 250
-		equilibrium_power = 250
-		icon_state = base_icon_state
+	var/turf/T = get_turf(src)
+	for(var/mob/M in GLOB.player_list)
+		var/turf/mob_turf = get_turf(M)
+		if(T.z == mob_turf.z)
+			SEND_SOUND(M, 'sound/magic/charge.ogg')
+
+			if(M.z == z)
+				to_chat(M, "<span class='boldannounce'>You feel reality distort for a moment...</span>")
+			else
+				to_chat(M, "<span class='boldannounce'>You hold onto \the [M.loc] as hard as you can, as reality distorts around you. You feel safe.</span>")
+
+	if(combined_gas > MOLE_PENALTY_THRESHOLD)
+		investigate_log("has collapsed into a singularity.", "supermatter")
```

Comment:
Preferably a `message_admins()` here to say `"A supermatter has delaminated and spawned a singularity (JMP)"`